Fix cm5 and 5b+ typec cannot find u disk.

Ken-Vamrs opened the pull request.

Ken-Vamrs committed:
arm64: dts: cm5 io: update type-c configuration

Signed-off-by: Ken Wang <ken@radxa.com>

Ken-Vamrs committed:
arm64: dts: rock 5b+: update type-c configuration

Signed-off-by: Ken Wang <ken@radxa.com>

Ken-Vamrs committed:
drivers typec: add fixed-typec-miss-data property

Add fixed-typec-miss-data property to fixed typec miss data.

Signed-off-by: Ken Wang <ken@radxa.com>

Ken-Vamrs committed:
usb: typec: add disable-clean-disconnection property

Do not force a clean disconnect.

Signed-off-by: Ken Wang <ken@radxa.com>

Ken-Vamrs committed:
arm64: dts: rock 5b: add fixed-typec-miss-data and disable-clean-disconnection property

Signed-off-by: Ken Wang <ken@radxa.com>

RadxaYuntian reviewed (commented):
5b的最后三个commit是为了实现什么目的？

Ken-Vamrs commented:
> 5b的最后三个commit是为了实现什么目的？

是5B原先有两个个commit影响了cm5/5b+ 的 typec usb识别，然后加两个属性给5b指定使用原先那两个修改

RadxaYuntian commented on drivers/usb/typec/tcpm/tcpm.c:
```diff
@@ -6064,6 +6081,8 @@ static int tcpm_fw_get_caps(struct tcpm_port *port,
 		port->typec_caps.data = ret;
 	}
 
+	fixed_typec_miss_data = fwnode_property_read_bool(fwnode, "fixed-typec-miss-data");
```
要修改struct tcpm_port的定义，把两个新增的变量加进去，而不是使用全局变量

Ken-Vamrs replied:
好的，已修改

Ken-Vamrs committed:
typec: tcpm: move some variables to tcpm_port struction

Move fixed_typec_miss_data and disable_clean_disconnection variables to tcpm_port struction

Signed-off-by: Ken Wang <ken@radxa.com>

RadxaYuntian commented on drivers/usb/typec/tcpm/tcpm.c:
```diff
@@ -4440,7 +4451,7 @@ static void run_state_machine(struct tcpm_port *port)
 		 * port->explicit_contract.
 		 */
-		if (port->explicit_contract && port->data_role == TYPEC_HOST)
+		if ((port->explicit_contract && !port->fixed_typec_miss_data) || (port->explicit_contract && port->data_role == TYPEC_HOST))
```
```suggestion
		if ((port->explicit_contract) && (!port->fixed_typec_miss_data || port->data_role == TYPEC_HOST))
```
这样可读性更好

RadxaYuntian commented on drivers/usb/typec/tcpm/tcpm.c:
```diff
@@ -5967,7 +5978,14 @@ static void tcpm_init(struct tcpm_port *port)
+
+	if (!port->fixed_typec_miss_data){
+		port->vbus_present = port->tcpc->get_vbus(port->tcpc);
+
```
```suggestion
```

Ken-Vamrs committed:
typec: tcpm: optimize the code to increase readability

Signed-off-by: Ken Wang <ken@radxa.com>
